src/server.js

```diff
@@ -56,100 +56,82 @@ http.createServer(function (request, response) {
-                        nextGoal = goal;
-                }
+    status.updated = new Date().toUTCString();
+}
 
-                status.earnings = campaign.attributes.pledge_sum / 100;
-                status.patrons = campaign.attributes.patron_count;
-                if (nextGoal)
-                    status.next_goal = nextGoal.attributes.amount_cents / 100;
-                status.updated = new Date().toUTCString();
-            } catch (err) {
-                console.log(err);
-            }
-        });
-    });
+async function refreshLiberapayStatus(account) {
+    const req = await fetch(process.env.LIBERAPAY_URL);
 
-    req.on("error", function(e) {
-        console.log("problem with request: " + e.message);
-    });
+    const res = await req.json();
 
-    req.end();
+    status["liberapayEarnings_" + account] = res.receiving.amount
+    status["liberapayPatrons_" + account] = res.npatrons
+    status["liberapayUpdated_" + account] = new Date().toUTCString();
 }
 
-function refreshLiberapayStatus(account) {
-    // https://liberapay.com/Tiled/public.json
-    var options = {
-        hostname: "liberapay.com",
-        path: "/" + account + "/public.json",
-        rejectUnauthorized: false,
-        agent: false,
-    };
-
-    var req = https.request(options, function(res) {
-        var body = ""
-
-        res.on("data", function(chunk) { body += chunk; });
-
-        res.on("end", function() {
-            try {
-                var json = JSON.parse(body);
-
-                status["liberapayEarnings_" + account] = json.receiving.amount
-                status["liberapayPatrons_" + account] = json.npatrons
-                status["liberapayUpdated_" + account] = new Date().toUTCString();
-            } catch (err) {
-                console.log(err);
-            }
-        });
-    });
+async function refreshGithubSponsors() {
 
-    req.on("error", function(e) {
-        console.log("problem with request: " + e.message);
+    const query = `
+    query {
+        user(login: "bjorn") {
+        ... on Sponsorable {
+                sponsors(first: 10) {
+                    totalCount
+                }
+            },
+            estimatedNextSponsorsPayoutInCents
+        }
+    }`
+
+    const req = await fetch(process.env.GITHUB_URL, {
+        method: 'POST',
+        body: JSON.stringify({query}),
+        headers: {
+            'Authorization': `Bearer ${process.env.GITHUB_TOKEN}`
+        }
     });
 
-    req.end();
+    const res = await req.json();
+
+    status['githubIncomeFromSponsors'] = ((res.data.estimatedNextSponsorsPayoutInCents ?? 0) / 100).toFixed(2)
+    status["githubSponsors"] = res.data.user.sponsors.totalCount;
 }
 
 function refreshStatus() {
     refreshPatreonStatus();
     refreshLiberapayStatus("Tiled");
     refreshLiberapayStatus("bjorn");
```

Comment:
Let's forget about checking multiple Liberapay accounts for now (this change breaks it currently, because the same URL is used, but we can simplify it to check only a single account).

Reply:
Which account do you want removed?